src/pool/srv_pool.c

```diff
@@ -6719,9 +6745,7 @@ ds_pool_ranks_get_handler(crt_rpc_t *rpc)
 		D_GOTO(out, rc = -DER_INVAL);
 
 	/* Get available ranks */
-	rc = ds_pool_get_ranks(in->prgi_op.pi_uuid,
-			       PO_COMP_ST_UP | PO_COMP_ST_UPIN | PO_COMP_ST_DRAIN,
-			       &out_ranks);
+	rc = ds_pool_get_ranks(in->prgi_op.pi_uuid, POOL_GROUP_MAP_STATUS, &out_ranks);
```

Comment:
Definitely no change requested in this PR. Perhaps in a future patch we can gracefully remove certain RPC definitions and "dead code" handlers such as this one. Perhaps the POOL_REPLICAS_ADD/REMOVE RPCs too?

Reply:
Actually I didn't even realize `POOL_RANKS_GET` is no longer used. Sure, code like this can be removed, though we might want to retain the RPC operation code (or, is that necessary)?

Reply:
Not sure whether the opcode retention is needed, as opposed to incrementing the protocol version number. Would seem safer to keep the opcode (but wouldn't that require a handler function anyway?). I'm not sure what other complications may be introduced by bumping the protocol version number (e.g., right now the code maintains 2 recent protocol version numbers 4 and 5). Definitely something we could look at separately IMO. I can file a ticket on it to work on in the future if you like.

Reply:
Sure, sounds good to me. Thanks, Ken.

Reply:
I filed https://daosio.atlassian.net/browse/DAOS-13038 for the unused opcodes/handlers issue